pkg/kapp/app/recorded_app.go

```diff
@@ -80,9 +83,31 @@ func (a *RecordedApp) UpdateUsedGVs(gvs []schema.GroupVersion) error {
 func (a *RecordedApp) CreateOrUpdate(labels map[string]string) error {
 	defer a.logger.DebugFunc("CreateOrUpdate").Finish()
 
+	if os.Getenv("USE_EXISTING_CONFIGMAP_NAME") != "True" {
+		configMap, err := a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.name, metav1.GetOptions{})
+		if err == nil {
+			err = a.renameConfigMap(configMap, a.fqName, a.nsName)
```

Comment:
I think we should check if the old ConfigMap was created by kapp or not, otherwise we might end up renaming some user ConfigMap. 
This could happen in the following scenario:

1. User creates a ConfigMap named `xyz`.
2. User tries to create a kapp application named `xyz` with `USE_EXISTING_CONFIGMAP_NAME` being set to False.

Although it would error out after that as no meta data would be found in the ConfigMap, but the damage would have been done by then.